base/tuple.jl

```diff
@@ -14,6 +14,11 @@ true
 """
 NTuple
 
+# convenience function for extracting N from a Tuple (if defined)
+# else return `nothing` for anything else given (such as Vararg or other non-sized Union)
+_counttuple(::Type{<:NTuple{N,Any}}) where {N} = N
```

Comment:
Should you have a `@nospecialize` on the argument in this case? I'm still not 💯% clear on it, but since we don't need to generate any more specializations/method-instances other than the two provided, should these be marked `@nospecialize`? (Or is that done automatically somehow given their tiny function bodies?)